src/sqlfluff/core/rules/functional/raw_file_slices.py

```diff
@@ -0,0 +1,34 @@
+"""Surrogate class for working with RawFileSlice collections."""
+from typing import Callable
+
+from sqlfluff.core.templaters.base import RawFileSlice, TemplatedFile
+
+
+class RawFileSlices(list):
+    """Encapsulates a sequence of one or more RawFileSlice.
+
+    The slices may or may not be contiguous in a file.
+    Provides useful operations on a sequence of slices to simplify rule creation.
+    """
+
+    def __new__(cls, templated_file, *raw_slices):
+        """Override new operator."""
+        return super(RawFileSlices, cls).__new__(cls, raw_slices)
+
+    def __init__(self, templated_file: TemplatedFile, *raw_slices: RawFileSlice):
+        super().__init__(raw_slices)
+        self.templated_file = templated_file
+
+    def all(self, *predicates: Callable[[RawFileSlice], bool]) -> bool:
+        """Do all the raw slices match?"""
+        for s in self:
+            if predicates and not any(p(s) for p in predicates):
+                return False
+        return True
```

Comment:
```suggestion
    def all(self, *predicates: Optional[Callable[[RawFileSlice], bool]] = None) -> bool:
        """Do all the raw slices match?"""
        for s in self:
            if predicates is not None and not any(p(s) for p in predicates):
                return False
        return True
```
this should be made consistent with `segments.py`

Reply:
Will do!